src/OrleansServiceBus/Providers/Streams/EventHub/IEventHubQueueCacheFactory.cs

```diff
@@ -0,0 +1,14 @@
+﻿using Orleans.Runtime;
+using Orleans.Streams;
+
+namespace Orleans.ServiceBus.Providers
+{
+    /// <summary>
+    /// Factory responsible for creating a message cache for an EventHub partition.
+    /// </summary>
+    public interface IEventHubQueueCacheFactory
```

Comment:
I personally don't like `Initialize` approach since in fact it violates encapsulation (every method call moves an object from one valid state to other valid state) - after ctor call but before `Initialize`, the object is in an inconsistent state. Then, it is mentioned in https://blogs.msdn.microsoft.com/orleans/2017/04/12/de-inventing-the-wheel/ that the goal is to move away from 2-step initialization